### PR TITLE
Add conditional focus to survey task Choice components

### PR DESCRIPTION
**Pull request by mcbouslog**
Package:
- lib-classifier

Towards #1991.

Staging:
- https://local.zooniverse.org:8080/?project=1634&workflow=2434

Describe your changes:
- conditionally adds focus to Choice components, as follows:
- Choice with more than one image focus (after 1 tab) on Carousel
- Choice with no images, yes confusions, focus on first confusion button
- Choice with no images or confusions, yes questions, focus on first question
  - if first question is checkbox then first input gets focus
  - if first question is radio and no existing answer then first input gets focus
  - if first question is radio and existing answer then existing answer gets focus
- Choice with no images, confusions, or questions focus is on "Identify" button

# Review Checklist

## General

- [x] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/Choice.js [hunk 1]
```diff
@@ -42,6 +51,7 @@ export default function Choice (props) {
     >
       {choice.images?.length > 0 && (
         <Carousel
+          autoFocus={hasFocus === 'images'}
```
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/Choice.js:54]: `autoFocus` isn't a property of the `Carousel`, this should be deleted. I wasn't able to determine a simple way to focus the Carousel if rendered, please let me know if something comes to mind.

Comment (srallen) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/Choice.js:54]: I had a look at the `Carousel` and under the hood, it passes all other props to a `Stack` and that under the hood is a styled div, so I don't think it'll work.

Check it out: https://github.com/grommet/grommet/blob/d4934e7d9749e67f953dcfb4a571b3a514d7d2e6/src/js/components/Carousel/Carousel.js#L178

If we want to focus this, we'll have to use javascript. I don't think we really need this to be auto-focusable. Getting to the inputs is more important.

**Review by srallen — CHANGES_REQUESTED** [review]
While trying out https://local.zooniverse.org:8080/?project=1634&workflow=2434 I found that I couldn't click or keyboard space/enter the "Not this" option of a choice. I'm not getting an errors related to it. The button doesn't appear to be disabled in the DOM and has an onClick event handler defined, so I'm not sure what's going on here. I checked the behavior on master, and it's broken there too, so this is out of scope for this branch, but something to be aware of.

As far as requested changes, I'd say we can remove the autofocus from the carousel as noted.

**Comment by mcbouslog** [comment]
I made a mistaken during a previous PR rebase and broke the "Not this" functionality, but include a fix in #2256 [here](https://github.com/zooniverse/front-end-monorepo/pull/2256/files#diff-440b4febc47c080a86bf7666488637387d0cd9cdae19300a515b5f58ca646036R34).

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/CheckBoxInputs/components/CheckBoxInput.js [hunk 2]
```diff
@@ -15,11 +15,16 @@ export const StyledBox = styled(Box)`
     opacity: 0.01;
     position: absolute;
   }
+
+  &:focus-within {
+    box-shadow: 0 0 2px 2px ${props => props.theme.global.colors['accent-1']};
```
Comment (srallen) [inline packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/CheckBoxInputs/components/CheckBoxInput.js:20]: Non-blocking: the theme has a designation for a focus color at `theme.global.colors.focus` which right now is mapping to `accent-1`, but if we change it, then this wouldn't need to be updated in that case.

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/RadioInputs/components/RadioInput.js [hunk 3]
```diff
@@ -15,11 +15,16 @@ export const StyledBox = styled(Box)`
     opacity: 0.01;
     position: absolute;
   }
+
+  &:focus-within {
+    box-shadow: 0 0 2px 2px ${props => props.theme.global.colors['accent-1']};
```
Comment (srallen) [inline packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/RadioInputs/components/RadioInput.js:20]: Same comment about using the focus attribute in the theme object.

**Review by srallen — APPROVED** [review]
There's some awkwardness with the focus scrolling to the focused input while the choice image is still loading, but it's not a blocker for me. I can explain more on call if needed.

File: packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/CheckBoxInputs/components/CheckBoxInput.js (hunk 2 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/SurveyTask/components/components/Choice/components/Questions/components/CheckBoxInputs/components/CheckBoxInput.js]: forgot about that, always good to set programmatically 👍

**Comment by mcbouslog** [comment]
> There's some awkwardness with the focus scrolling to the focused input while the choice image is still loading, but it's not a blocker for me. I can explain more on call if needed.

Added a note about this to #1991 to revisit shortly.

**Pull request merged**